spec/datadog/tracing/trace_operation_spec.rb

```diff
@@ -2187,6 +2187,106 @@ def span
   end
 
   describe 'integration tests' do
+    context 'service_entry attributes' do
+      context 'when service not given' do
+        it do
+          trace_op.measure('root') do |_, trace|
+            trace.measure('children_1') do
+              # sleep(0.01)
+            end
+
+            trace.measure('children_2') do
+              # sleep(0.01)
+            end
+          end
+
+          trace_segment = trace_op.flush!
+
+          expect(trace_segment.spans.map(&:service)).to all(be_nil)
+
+          hash = trace_segment.spans.each_with_object({}) do |span, h|
```

Comment:
Minor: Hash usage is kind of ugly, but I do understand that it provides an easier mapping. Would it be an improvement to put the logic in some sort of RSpec matcher?

```
expect(trace_segment.spans).to include(
  a_span_with(name: 'root', service_entry: true),
  a_span_with(name: 'children_1', service_entry: false),
  a_span_with(name: 'children_2', service_entry: false)
)
```